### PR TITLE
Move all base class calls to the start of their method.

### PR DESCRIPTION
**Pull request by NogginBops**
And remove all resource cleanup and added an explanation as to why it shouldn't be done and explained appropriate situations for resource cleanup.

### CODE REVIEW

File: Chapter1/2-HelloTriangle/Window.cs [hunk 1]
```diff
@@ -160,33 +162,46 @@ protected override void OnRenderFrame(FrameEventArgs e)
...
+        // should not be done when exiting the application.
+        //
+        // Places where cleanup is appropriate would be to delete textures that are no
+        // longer used for whatever reason (e.g. a new scene is loaded that doesn't use a texture).
+        // This would free up video ram (VRAM) that can be used for new textures.
+        //
+        // The comming chapters will not have this code.
         protected override void OnUnload()
         {
+            base.OnUnload();
```
Comment (deccer) [inline Chapter1/2-HelloTriangle/Window.cs:203]: I think in `Unload` it makes sense to keep it at the end of the method, as you are unwinding all resources in the opposite order than `Load`

**Review by deccer — CHANGES_REQUESTED** [review]
Just one little thing, otherwise it looks ok to me

File: Chapter1/2-HelloTriangle/Window.cs (hunk 1 above)
Comment (PJB3005) [reply Chapter1/2-HelloTriangle/Window.cs:203]: At least in the case of Unload there's no actual code running in the supercall (outside of firing events) so I'd say the order doesn't really matter.

Comment (deccer) [reply Chapter1/2-HelloTriangle/Window.cs:203]: I am aware, but for consistency reasons and to show newcomers

**Review by deccer — APPROVED** [review]
Thanks :)

**Pull request merged**